tm2py/components/demand/commercial.py

```diff
@@ -0,0 +1,611 @@
+        # note that very small, small and medium are assigned as one class
+        # and share the same output skims
+        ffactors = self._load_ff_lookup_tables()
+        k_factors = self._load_k_factors()
+        friction_calculations = [
+            {"name": "vsmtrk", "time": trk_blend_time, "use_k_factors": False},
+            {"name": "smltrk", "time": trk_blend_time, "use_k_factors": True},
+            {"name": "medtrk", "time": trk_blend_time, "use_k_factors": True},
+            {"name": "lrgtrk", "time": lrgtrk_blend_time, "use_k_factors": True},
+        ]
+        daily_demand = {}
+        for spec in friction_calculations:
+            name = spec["name"]
+            # lookup friction factor values from table with interpolation
+            # and multiply by k-factors (no k-factors for very small truck)
+            friction_matrix = np.interp(spec["time"], ffactors["time"], ffactors[name])
+            if spec["use_k_factors"]:
+                friction_matrix = friction_matrix * k_factors
+            # run matrix balancing
+            prod_attr_matrix = self._matrix_balancing(
+                friction_matrix,
+                trip_ends[f"{name}_prod"].to_numpy(),
+                trip_ends[f"{name}_attr"].to_numpy(),
+                name,
+            )
+            daily_demand[name] = (
+                0.5 * prod_attr_matrix + 0.5 * prod_attr_matrix.transpose()
+            )
+            self.logger.log(
+                f"{name}, prod sum: {prod_attr_matrix.sum()}, "
+                f"daily sum: {daily_demand[name].sum()}",
+                level="DEBUG",
+            )
+        return daily_demand
+
+    def _get_blended_time(self, name: str) -> NumpyArray:
+        """Blend AM and MD truck times for distribution calculations.
+
+        Uses 1/3 * AM + 2/3 * MD, sources skims from generated OMX files
+        """
+        skim_path_tmplt = self.get_abs_path(self.config.highway.output_skim_path)
+        with OMXManager(skim_path_tmplt.format(period="AM"), "r") as am_file:
+            am_time = am_file.read(f"am_{name}_time")
+        with OMXManager(skim_path_tmplt.format(period="MD"), "r") as md_file:
+            md_time = md_file.read(f"md_{name}_time")
+        # blend truck times
+        # compute blended truck time as an average 1/3 AM and 2/3 MD
+        #     NOTE: Cube outputs skims\COM_HWYSKIMAM_taz.tpp, skims\COM_HWYSKIMMD_taz.tpp
+        #           are in the highway_skims_{period}.omx files in Emme version
+        #           with updated matrix names, {period}_trk_time, {period}_lrgtrk_time.
+        #           Also, there will no longer be separate very small, small and medium
+        #           truck times, as they are assigned together as the same class.
+        #           There is only the trk_time.
+        blend_time = 1 / 3.0 * am_time + 2 / 3.0 * md_time
+        return blend_time
+
+    @LogStartEnd(level="DEBUG")
+    def _time_of_day(
```

Comment:
this function name should  be an action, not a noun - ex: `_split_by_time_period